hms-app/src/components/HackathonSelectDropdown.tsx

```diff
@@ -1,22 +1,33 @@
 import React, { useEffect, useState } from 'react'
 import { getListOfHackathons } from '../actions/HackathonActions'
-import { Select } from '@mantine/core'
+import { Select, SelectItem } from '@mantine/core'
 import { HackathonPreview } from '../common/types'
 
 type Props = {
-  setHackathonID: (hackthonID: string) => void
+  setHackathonId: (hackthonID: string) => void
+  context: HackathonDropdownMode
 }
 
-export default function HackathonSelectDropdown(setHackathonID: Props) {
-  const [selectedHackweek, setSelectedHackweek] = useState('')
+export enum HackathonDropdownMode {
+  Archive = 'ARCHIVE',
+  IdeaPortal = 'IDEA_PORTAL',
+  Home = 'HOME',
+  YourIdeas = 'YOUR_IDEAS',
+}
+
+export default function HackathonSelectDropdown({
+  setHackathonId,
+  context,
+}: Props) {
   const [isError, setIsError] = useState(false)
   const [isLoading, setIsLoading] = useState(true)
-  const [hackathonList, setHackathonList] = useState([] as HackathonPreview[])
+  const [hackathonList, setHackathonList] = useState<HackathonPreview[]>([])
+  const today = new Date().setHours(0, 0, 0, 0)
```

Comment:
the docs say that `setHours` will return the number of millis since the epoch, so `today` won't be a date anymore.  You should do something more like...
```suggestion
  const today = new Date();
  today.setHours(0, 0, 0, 0)
```